lib/rucio/core/replica.py

```diff
@@ -3484,32 +3891,40 @@ def get_suspicious_files(rse_expression, available_elsewhere, filter_=None, logg
         for rse in parsedexp:
             rse_clause.append(models.RSEFileAssociation.rse_id == rse['id'])
 
-    # query base
-    query = session.query(func.count(), bad_replicas_alias.scope, bad_replicas_alias.name, models.RSEFileAssociation.rse_id, func.min(models.RSEFileAssociation.created_at))\
-                   .filter(models.RSEFileAssociation.rse_id == bad_replicas_alias.rse_id,
-                           models.RSEFileAssociation.scope == bad_replicas_alias.scope,
-                           models.RSEFileAssociation.name == bad_replicas_alias.name,
-                           bad_replicas_alias.created_at >= younger_than)
+    stmt = select(
+        func.count(),
+        bad_replicas_alias.scope,
+        bad_replicas_alias.name,
+        models.RSEFileAssociation.rse_id,
+        func.min(models.RSEFileAssociation.created_at)
+    ).select_from(
+        bad_replicas_alias
+    ).where(
+        models.RSEFileAssociation.rse_id == bad_replicas_alias.rse_id,
+        models.RSEFileAssociation.scope == bad_replicas_alias.scope,
+        models.RSEFileAssociation.name == bad_replicas_alias.name,
+        bad_replicas_alias.created_at >= younger_than
+    )
     if is_suspicious:
-        query.filter(bad_replicas_alias.state == BadFilesStatus.SUSPICIOUS)
+        stmt = stmt.where(bad_replicas_alias.state == BadFilesStatus.SUSPICIOUS)
     if rse_clause:
-        query = query.filter(or_(*rse_clause))
+        stmt = stmt.where(or_(*rse_clause))
 
     # Only return replicas that have at least one copy on another RSE
     if available_elsewhere == SuspiciousAvailability["EXIST_COPIES"].value:
         available_replica = exists(select(1).where(and_(replicas_alias.state == ReplicaState.AVAILABLE,
                                                         replicas_alias.scope == bad_replicas_alias.scope,
                                                         replicas_alias.name == bad_replicas_alias.name,
                                                         replicas_alias.rse_id != bad_replicas_alias.rse_id)))
-        query = query.filter(available_replica)
+        stmt = stmt.where(available_replica)
 
     # Only return replicas that are the last remaining copy
     if available_elsewhere == SuspiciousAvailability["LAST_COPY"].value:
         last_replica = ~exists(select(1).where(and_(replicas_alias.state == ReplicaState.AVAILABLE,
```

Comment:
Nitpick. (Well all of these are nitpicks.) Can the `.where(....` get stacked under the select?

Reply:
Sure, changed all of these queries to stack the `where`s under `select(1)`s, like so:

```
    if available_elsewhere == SuspiciousAvailability["LAST_COPY"].value:
        last_replica = ~exists(select(1)
                               .where(and_(replicas_alias.state == ReplicaState.AVAILABLE,
                                           replicas_alias.scope == bad_replicas_alias.scope,
                                           replicas_alias.name == bad_replicas_alias.name,
                                           replicas_alias.rse_id != bad_replicas_alias.rse_id)))
```